src/lib.rs

```diff
@@ -171,29 +171,31 @@ pub use crate::spooled::{spooled_tempfile, SpooledData, SpooledTempFile};
 
 /// Create a new temporary file or directory with custom parameters.
 #[derive(Debug, Clone, Eq, PartialEq)]
-pub struct Builder<'a, 'b> {
+pub struct Builder<'a, 'b, 'c> {
```

Comment:
This is a breaking change. IMO, I messed up and should have had just _one_ lifetime here (relying on sub-typing) but fixing that would also be a breaking change.

We can do one of:

1. Re-use one of the other lifetimes.
2. Store the directory by-value (although I'd still take the parameter as `AsRef`).

I'd go with the former approach (probably using the prefix's lifetime, `'a`. It's a bit awkward but it provides the best upgrade path for an eventual v4.